File: Algorithms/Implementation/Subarray Division.js
// Subarray Division

// https://www.hackerrank.com/challenges/the-birthday-bar/problem

function birthday(s, d, m) {
  // Write your code here
  let days = 0
  let count = 0
  for (let i = 0; i < m; i++) {
    days += s[i]
  }

  let l = 0
  let r = m

  while (r <= s.length) {
    if (days === d) {
      count++
    }
    days = days - s[l] + s[r]
    l++
    r++
  }

  return count
}

File: Algorithms/Implementation/Subarray Division.py
# Subarray Division

def birthday(s, d, m):
  # Write your code here
  count = 0
  days = 0
  for i in range(m):
    days += s[i]

  if days == d:
    count += 1

  l, r = 0, m

  while r < len(s):
    days = days - s[l] + s[r]
    if days == d:
      count += 1
    l += 1
    r += 1

  return count